packages/plugin-ext/src/main/browser/debug/debug-main.ts

```diff
@@ -178,6 +181,27 @@ export class DebugMainImpl implements DebugMain, Disposable {
         }
     }
 
+    $registerDebugConfigurationProvider(description: DebugConfigurationProviderDescriptor): void {
+        const handle = description.handle;
+        const toDispose = new DisposableCollection(
+            Disposable.create(() => this.configurationProviders.delete(handle))
+        );
+        this.configurationProviders.set(handle, toDispose);
+
+        toDispose.push(
+            this.pluginDebugService.registerDebugConfigurationProvider(new PluginDebugConfigurationProvider(description, this.debugExt))
+        );
+
+        this.toDispose.push(Disposable.create(() => this.$unregisterDebugConfigurationProvider(handle)));
```

Comment:
Perhaps `this.toDispose.push(toDispose)`? If it's disposed via an external call to `$unregisterDebugConfigurationProvider`, it will also be removed from `this.toDIspose`, so I believe the effect is the same.